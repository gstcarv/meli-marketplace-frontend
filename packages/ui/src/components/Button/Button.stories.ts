import type { Meta, StoryObj } from '@storybook/react';
import { Button } from './Button';

const meta = {
    title: 'core/Button',
    component: Button,
    parameters: {
        layout: 'centered'
    },
    tags: ['autodocs']
} satisfies Meta<typeof Button>;

export const Primary: StoryObj<typeof meta> = {
    args: {
        children: 'Click me'
    }
};

export default meta;
