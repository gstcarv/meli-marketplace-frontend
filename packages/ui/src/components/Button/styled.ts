import styled from 'styled-components';

export const ButtonBase = styled.button`
    height: 48px;
    border: none;
    padding: 0 ${({ theme }) => theme.spacing.get('xl2')};
    gap: 0 ${({ theme }) => theme.spacing.get('sm')};
    border-radius: ${({ theme }) => theme.radii.get('md')};
    background-color: ${({ theme }) => theme.colors.get('accent-500')};
    transition: background-color 0.8s;
    display: flex;
    justify-content: center;
    align-items: center;
    cursor: pointer;

    &:hover {
        background-color: ${({ theme }) => theme.colors.get('accent-600')};
    }
`;

export const ButtonText = styled.span`
    font-size: ${({ theme }) => theme.fontSize.get('body')};
    font-weight: ${({ theme }) => theme.fontWeight.get('bold')};
    color: #ffffff;
    letter-spacing: 0.3px;
`;
