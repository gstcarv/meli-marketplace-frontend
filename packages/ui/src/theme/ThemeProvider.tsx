import 'react-loading-skeleton/dist/skeleton.css';
import { PropsWithChildren } from 'react';
import { ThemeProvider as StyledThemeProvider } from 'styled-components';
import { GlobalStyle } from '../globalStyles';
import { theme } from './theme';

export const ThemeProvider = (props: PropsWithChildren) => (
    <StyledThemeProvider theme={theme}>
        <>
            {props.children}

            <GlobalStyle />
        </>
    </StyledThemeProvider>
);
