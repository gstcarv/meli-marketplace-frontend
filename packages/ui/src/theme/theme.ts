import { colors, fontSize, radii, spacing, fontWeight } from '../tokens';

export const theme = { colors, fontSize, radii, spacing, fontWeight };

export type Theme = typeof theme;
