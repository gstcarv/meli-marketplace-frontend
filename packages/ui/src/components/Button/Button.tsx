import { ComponentProps } from 'react';
import { ButtonBase, ButtonText } from './styled';
import { ButtonProps } from './types';

export const Button = ({ children, ...props }: ButtonProps) => {
    return (
        <ButtonBase {...(props as ComponentProps<typeof ButtonBase>)}>
            <ButtonText>{children}</ButtonText>
        </ButtonBase>
    );
};
