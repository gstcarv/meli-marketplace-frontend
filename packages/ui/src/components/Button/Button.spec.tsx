import { fireEvent, render, screen } from '@testing-library/react';
import { ThemeProvider } from '../../theme';
import { Button } from './Button';

describe('<Button />', () => {
    it('should render button correctly', () => {
        const { container } = render(
            <ThemeProvider>
                <Button>my button content</Button>
            </ThemeProvider>
        );

        expect(screen.getByText('my button content'));

        expect(container).toMatchSnapshot();
    });

    it('should call onClick on click button', () => {
        const onClickFn = jest.fn();

        render(
            <ThemeProvider>
                <Button onClick={onClickFn}>click me</Button>
            </ThemeProvider>
        );

        fireEvent.click(screen.getByRole('button', { name: 'click me' }));

        expect(onClickFn).toHaveBeenCalledTimes(1);
    });
});
