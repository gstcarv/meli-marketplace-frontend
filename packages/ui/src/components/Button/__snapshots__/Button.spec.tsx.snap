// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render button correctly 1`] = `
<div>
  <button
    class="sc-blHHSb hAuOKQ"
  >
    <span
      class="sc-gtLWhw euxXeq"
    >
      my button content
    </span>
  </button>
</div>
`;
